documentation/CAMARA-ICM-examples.md

```diff
@@ -0,0 +1,128 @@
+This document does not say that the access token is self-contained or not.
+
+---
+
+### Purpose as a scope: Requesting one purpose, two scopes of the same API
+
+#### OIDC authorization code flow with one purpose as scope
+
+See [OIDC Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)
+
+```
+GET /authorize?
+    response_type=code
+    &scope=openid%20dpv%3AFraudPreventionAndDetection%20sim-swap%3Acheck%20sim-swap%3Aretrieve-date
+    &client_id=s6BhdRkqt3
+    &state=af0ifjsldkj
+    &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb HTTP/1.1
+Host: server.example.com
+```
+
+#### Successful response redirecting the user agent 
+
+See [OIDC Successful Authentication Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse)
+
+(with line wraps within values for display purposes only)
+
+```
+HTTP/1.1 302 Found 
+Location: https://client.example.com/cb?code=SplxlOBeZQQYbYS6WxSbIA&state=af0ifjsldkj
+```
+
+#### Access token request
+
+See [OIDC Token Request](https://openid.net/specs/openid-connect-core-1_0.html#TokenRequest)
+
+
+```
+POST /token HTTP/1.1
+
+Host: server.example.com 
+Content-Type: application/x-www-form-urlencoded 
+
+grant_type=authorization_code&code=SplxlOBeZQQYbYS6WxSbIA
+    &redirect_uri=https%3A%2F%2Fclient%2Eexample%2Ecom%2Fcb
+    &client_assertion_type=urn%3Aietf%3Aparams%3Aoauth%3Aclient-assertion-type%3Ajwt-bearer
+    &client_assertion=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3Mi......
+```
+
+#### Successful response
+
+See [OIDC Successful Token Response](https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse)
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+{
+  "access_token": "SlAV32hkKG",
+  "token_type": "Bearer",
+  "refresh_token": "8xLOxBtZp8",
+  "expires_in": 3600,
+  "id_token": "eyJhbGciOiJSUz....",
+  "scope": "openid dpv:FraudPreventionAndDetection sim-swap:check sim-swap:retrieve-date"
+}
+```
+
+#### RFC9101 request object with one purpose as scope
+
+```
+{
+  "iss": "s6BhdRkqt3",
+  "aud": "https://server.example.com",
+  "response_type": "code",
+  "client_id": "s6BhdRkqt3",
+  "redirect_uri": "https://client.example.org/cb",
+  "scope": "openid dpv:FraudPreventionAndDetection sim-swap:check sim-swap:retrieve-date",
+  "state": "af0ifjsldkj",
+  "nonce": "n-0S6_WzA2Mj",
+  "max_age": 86400,
+  "exp": 1419356238,
+  "iat": 1419350238
+}
+```
+
+#### CIBA authentication request with one purpose and two scopes
+
+See [CIBA authentication request](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.7.1)
+
+```
+POST /bc-authorize HTTP/1.1
+   Host: server.example.com
+   Content-Type: application/x-www-form-urlencoded
+
+   scope=openid%20dpv%3AFraudPreventionAndDetection%20sim-swap%3Acheck%20sim-swap%3Aretrieve-date&
+   client_notification_token=8d67dc78-7faa-4d41-aabd-67707b374255&
```

Comment:
Isn't client_notification_token only needed for ping and push modes?

Reply:
Indeed, it is used in push and ping mode to authenticate when notifying the "Endpoint Notification Client"

Reply:
fixed